src/Catty/ViewController/Download/Details/ProjectDetailStoreViewControllerExtension.swift

```diff
@@ -0,0 +1,411 @@
+            reportButton.setTitleColor(UIColor.buttonTint, for: .normal)
+            if let view = view {
+                view.addSubview(reportButton)
+                setMaxHeightIfGreaterFor(view, withHeight: view.frame.size.height + reportButton.frame.size.height)
+            }
+        }
+    }
+
+    func addInformationLabel(to view: UIView?, withAuthor author: String?, downloads: NSNumber?, uploaded: String?, version: String?, views: NSNumber?) {
+        var version = version
+        let height = self.height()
+        var offset = (view?.frame.size.height ?? 0.0) + height * 0.05
+        self.addHorizontalLine(to: view, andHeight: offset - 15)
+        let informationLabel = UILabel(frame: CGRect(x: (view?.frame.size.width ?? 0.0) / 15, y: offset, width: 155, height: 25))
+        informationLabel.text = kLocalizedInformation
+        self.configureTitleLabel(informationLabel, andHeight: height)
+        view?.addSubview(informationLabel)
+        offset += height * 0.075
+
+        version = version?.replacingOccurrences(of: "&lt;", with: "")
+        version = version ?? "" + (" MB")
+
+        let informationArray = [views?.stringValue ?? "", uploaded, version, downloads?.stringValue ?? ""]
+        let informationTitleArray = [UIImage(named: "viewsIcon"), UIImage(named: "timeIcon"), UIImage(named: "sizeIcon"), UIImage(named: "downloadIcon")]
+        var counter = 0
+        for info in informationArray {
+            let titleIcon = getInformationTitleLabel(
+                withTitle: informationTitleArray[counter],
+                atXPosition: (view?.frame.size.width ?? 0) / 12,
+                atYPosition: offset,
+                andHeight: height)
+            if let titleIcon = titleIcon {
+                view?.addSubview(titleIcon)
+            }
+
+            let infoLabel = getInformationDetailLabel(
+                withTitle: info,
+                atXPosition: (view?.frame.size.width ?? 0) / 12 + 25,
+                atYPosition: offset,
+                andHeight: height)
+            if let infoLabel = infoLabel {
+                view?.addSubview(infoLabel)
+            }
+
+            offset += +height * 0.04
+            counter += 1
+        }
+        setMaxHeightIfGreaterFor(view, withHeight: offset)
+    }
+
+    func addLoadingButton(to view: UIView?, withTarget target: Any?) {
+        let button = EVCircularProgressView()
+        button.tag = Int(kStopLoadingTag)
+        button.tintColor = UIColor.buttonTint
+        let xPosition = (view?.frame.size.width ?? 0.0) - 40
+        let yPosition = (view?.frame.size.height ?? 0.0) * 0.1 + Util.screenHeight() / 4.5
+        button.frame = CGRect(x: xPosition, y: yPosition - 25, width: 28, height: 28)
+        button.isHidden = true
+        button.addTarget(target, action: #selector(URLProtocol.stopLoading), for: .touchUpInside)
+        view?.addSubview(button)
+    }
+
+    func addDownloadAgainButton(to view: UIView?, withTarget target: Any?) {
```

Comment:
The `view` parameter should not be optional. Also, set the scope to `fileprivate`.